### PR TITLE
Prepare code to remove `GA` from Web and AMP

### PR DESCRIPTION
**Pull request by deedeeh**
## What does this change?

This PR removes Google Analytics from [theguardian.com](https://www.theguardian.com/uk) domain but not any other subdomains. Here is a list of the subdomains:

-  https://support.theguardian.com/uk/contribute
- https://workforus.theguardian.com/
- https://www.guardianjobsrecruiter.co.uk/

After this PR changes we shouldn't see:

- `window.gaData`
- `window.googleAnalyticsObject`
- `window.guardian.GAData` (articles)
-  `window.guardian.config.googleAnalytics`

We still have `window.gaGlobal` which is loaded by Google ads and we have no control to remove it.

This was tested locally and no requests are made to [google-analytics](https://www.google-analytics.com) 

Here are the other PRs related to this work https://github.com/guardian/commercial/pull/1393 and https://github.com/guardian/dotcom-rendering/pull/11470/

## Checklist

- [x] Tested locally, and on CODE if necessary
- [x] Will not break dotcom-rendering

### CODE REVIEW

**Comment by deedeeh** [comment]
This is the first attempt for a PR to prepare code to remove Google Analytics from the main domain theguardian.com for Web and AMP. I deleted files that I shouldn't delete as I believe it includes code for the whole Guardian estate.  I decided to start fresh with a new PR.

File: docs/02-architecture/06-client-side-architecture.md
Comment (deedeeh) [inline docs/02-architecture/06-client-side-architecture.md:156]: I tried to check if we have any other analytics for omniture or comscore but none of them in `base.scala.html` so I removed them but please correct me if I missed something!

File: static/src/javascripts/projects/common/modules/analytics/interaction-tracking.js
Comment (deedeeh) [inline static/src/javascripts/projects/common/modules/analytics/interaction-tracking.js:32]: Not sure if other code needs to be deleted in this file. I kept the code as it is used for Omniture as well unless someone thinks I should do something differently.

Comment (mxdvl) [reply static/src/javascripts/projects/common/modules/analytics/interaction-tracking.js:32]: I don't this we use Omniture anymore, but in any case it's best to remove in a separate PR.

File: common/app/views/fragments/inlineJSNonBlocking.scala.html
Comment (mxdvl) [inline common/app/views/fragments/inlineJSNonBlocking.scala.html:28]: Are you certain that this is only doing work related to GA? We would not want to lose Ophan tracking for this page.

File: common/app/model/meta.scala
Comment (mxdvl) [inline common/app/model/meta.scala:914]: Are you sure this is only relevant because to GA? It seems like it could be used by other first or third party scripts.

**Review by mxdvl — COMMENTED** [review]
In general this looks good, but I would urge caution when removing such a large part of the code base to make sure no other systems break, especially Ophan. Have you been able to deploy this on CODE and see events firing correctly?

**Comment by deedeeh** [comment]
> In general this looks good, but I would urge caution when removing such a large part of the code base to make sure no other systems break, especially Ophan. Have you been able to deploy this on CODE and see events firing correctly?

I deployed to CODE and I was only checking if we send any requests to GA but I haven't tracked all events. I shared this PR in the Ophan chat and @SHession  said: "A cursory glance suggest it's not impacting Ophan tracking, just GA and it's references to it. If that's the case I'm not worried about it affecting Ophan but do let me know if I've missed anything."

I am not sure which events exactly I need to check are firing correctly?

If you have any suggestions how this could be tested again to make sure nothing else is affected please share it with me.

File: applications/app/views/videoEmbed.scala.html
Comment (deedeeh) [reply applications/app/views/videoEmbed.scala.html:82]: I removed this code based on this https://github.com/guardian/frontend/pull/13798. @SHession could you confirm that this won't affect Ophan?

File: common/app/conf/switches/ABTestSwitches.scala
Comment (deedeeh) [reply common/app/conf/switches/ABTestSwitches.scala:44]: I had to change that because the build was failing but I will do a merge now.

File: static/src/javascripts/projects/common/modules/atoms/youtube-player.ts
Comment (deedeeh) [reply static/src/javascripts/projects/common/modules/atoms/youtube-player.ts:225]: This was just used by GA and I didn't find any references to it in the repo.

File: common/app/model/content.scala
Comment (deedeeh) [reply common/app/model/content.scala]: I am trying to remember why I removed this code in the first place but I really can't! That code shouldn't be touched. Addressed in 491f0d916a0c9ebdfcfe5a5e5ad122cc5137b9a1

File: common/app/model/meta.scala
Comment (deedeeh) [reply common/app/model/meta.scala]: GA might be using it but you are right it might be used elsewhere so I added it back the code in this file 491f0d916a0c9ebdfcfe5a5e5ad122cc5137b9a1 except for `normalisedContentType` which I think it could be used not just for GA as well?

File: common/app/views/fragments/inlineJSNonBlocking.scala.html
Comment (deedeeh) [reply common/app/views/fragments/inlineJSNonBlocking.scala.html]: Not sure if this code was just added for GA or it is used by Ophan tracking. @SHession could you confirm if this line of code is needed for Ophan tracking?

File: package.json
Comment (mxdvl) [inline package.json:32]: FYI, I’m inclined not to approve this PR untilthis point to an non-canary version to prevent an inadvertent release.

Comment (deedeeh) [reply package.json:32]: I understand. I am waiting until I get an approval from Commercial, merge and then bump the version in here. It is good to make this PR ready for an approval after addressing all the comments and merge Commercial.

File: common/app/views/fragments/inlineJSNonBlocking.scala.html
Comment (SHession) [reply common/app/views/fragments/inlineJSNonBlocking.scala.html]: It's difficult for me to say if this is needed for tracking as I'm not familiar with this codebase. However, it does look Ophan and not GA related and I would be inclined to keep it unless we are confident it won't impact traffic.

File: applications/app/views/videoEmbed.scala.html
Comment (SHession) [reply applications/app/views/videoEmbed.scala.html:82]: The comment suggests this is only needed for GA, but as with https://github.com/guardian/frontend/pull/27188#discussion_r1627117033, we will want to be as confident as possible that this is the case. Let me know if you need any help with checking.

File: static/src/javascripts/boot.js
Comment (akinsola-guardian) [inline static/src/javascripts/boot.js:60]: This probably is not needed anymore as it was used to determine the willShow param. 

willShowPrivacyMessage returns promise that resolves to `true` if the CMP will show the initial privacy message once it has initialised, or `false` if not.

Comment (deedeeh) [reply static/src/javascripts/boot.js:60]: I just want to be clear on this. Do you mean the whole conditional statement or just line 60?

Comment (akinsola-guardian) [reply static/src/javascripts/boot.js:60]: Sorry, just line 60.

Comment (deedeeh) [reply static/src/javascripts/boot.js]: It's ok I just wanted to be double check before I remove anything. Addressed in 672b13705366f884f084f722cce7888b504e79ec

File: applications/app/views/videoEmbed.scala.html
Comment (deedeeh) [reply applications/app/views/videoEmbed.scala.html]: Re-added in fadbbc5070e1a7655c478f8d7382ca7f97828333

File: common/app/views/fragments/inlineJSNonBlocking.scala.html
Comment (deedeeh) [reply common/app/views/fragments/inlineJSNonBlocking.scala.html]: Re-added in 491f0d916a0c9ebdfcfe5a5e5ad122cc5137b9a1

File: package.json
Comment (deedeeh) [reply package.json]: Done 420cd3bf2ec92e144cc0eafc014f5d4733da3e76

File: static/src/javascripts/boot.js
Comment (mxdvl) [inline static/src/javascripts/boot.js:59]: not a blocker, but I do think the containing code block can also be removed, as `recordedConsentTime` is not used anywhere else.

**Review by mxdvl — APPROVED** [review]
beGone anAlytics

File: static/src/javascripts/boot.js
Comment (deedeeh) [reply static/src/javascripts/boot.js:59]: That makes sense. I am not sure if the comment is up to date or not cause I am trying to find `App.tsx` in Frontend but can't find it!

I understand that I will remove the variable `recordedConsentTime` and the conditional statement that checks it. Please correct me if I missed something.

File: static/src/javascripts/boot.js
Comment (deedeeh) [reply static/src/javascripts/boot.js]: @mxdvl Also this conv. happened and I asked if I should remove the whole condition but told to remove just line 60.

File: static/src/javascripts/boot.js
Comment (mxdvl) [reply static/src/javascripts/boot.js:59]: I think it should have said `App.tsx` in `dotcom-rendering`

**Pull request merged**
